src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ClusterDetailsEventProcessor.java

```diff
@@ -40,6 +40,19 @@
    */
   private volatile ImmutableList<NodeDetails> nodesDetails = null;
 
+  public ClusterDetailsEventProcessor() {}
+
+  public ClusterDetailsEventProcessor(final ClusterDetailsEventProcessor other) {
+    if (other.nodesDetails != null) {
+      List<NodeDetails> nodeDetails = new ArrayList<>();
```

Comment:
Nit: use a builder for this instead of an alloc + copy
```
ImmutableList.Builder builder = new ImmutableList.Builder<NodeDetails>();
for (final NodeDetails oldDetails : other.nodesDetails) {
        builder.add(new NodeDetails(oldDetails));
}
this.nodesDetails = builder.build();
```

Reply:
Thanks a lot ! good point